### ISSUE TITLE
How to use new blocks instead of "legacy" blocks

### ISSUE THREAD

**Issue by scrobbleme** [woocommerce/woocommerce-gutenberg-products-block#6117]
Hello,

I'm currently building a new store using Blocktheme, which really works great and sooo much easier ...

Unfortunately there are too many differents between the legacy blocks (different gaps, colors...) and the new blocks.

So my questions are:

- Is it already possible to use the new blocks for archives?
   - (the easiest: I just replaces the legacy product archive block with the new All products block)
- When yes -> how?

Thanks

Adrian

**Comment by tomasztunik** [comment]
Hi Adrian, 

We are still in the process of getting everything ready for fully flashed out Full Site Editing experience. Updates to how the archive, single product and other built in templates and their blocks work is going to change completely with full control over the layout and composition of elements. 

For the time being if you wanted to push forward with blocks we'll love to hear feedback from you how it goes and what is missing but when you install blocks as standalone plugin this is "edge" version and will change a lot and as we wrap up work on the core, it will change quick.

Regarding the archives, when I was setting something up for testing I went with your approach as well. Unfortunately despite it looking like it is what we want it's not exactly the same. Things like built in filters and ordering based on URL doesn't work etc.

We'll be updating these to work with the QueryLoop soon and then it should be possible.

So much in the pipeline and we can't wait to get it out - we also agree that building with blocks completely transforms how we'll be able to build new pages/stores.

**Comment by scrobbleme** [comment]
Hello Tomasz,

Thanks for your answer.

As of now, my  major problem is the too many visual differences... 

- basically I have to style at least 2 types of product grids now (new and legacy), I also have seen, that even the new blocks are not totally the same (All products seems to be little different too)

-> at first my proposel would be:

- consolidate the styles (same typo, gaps, colors
- add some theme supprt "Use new blocks within legacy block" -> which just renders the new block within the legacy block instead


(Beside this, I don't want to bloat the tracker here with new issues, as there are already a lot and most of my problems are probably already there ;)

**Comment by tomasztunik** [comment]
We feel you on the styles difference, we could hack something quickly to make it more uniform but we want to do it proper when we do the rewrite of these listing components to use QueryLoop. We've started adding support for Global Styles and FSE theming which adds extra layers of complexity and cost. Because of that we don't plan to update theses styles in the short term.

But they will all be aligned once we update all the products list/grid blocks and templates! :) 

For the time being the only way to go about is to create CSS overrides. We made assumption that ultimately in most cases one would adjust these to their design regardless.
